leakcanary-android-core/src/test/java/leakcanary/LeakCanaryConfigTest.kt

```diff
@@ -0,0 +1,51 @@
+package leakcanary
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import kotlin.reflect.full.memberFunctions
+import kotlin.reflect.full.memberProperties
+
+class LeakCanaryConfigTest {
+
+  private fun configBuilderFunctions() = LeakCanary.Config.Builder::class.memberFunctions
+      .map { it.name }
+
+  private fun configProperties() = LeakCanary.Config::class.memberProperties
+      .filter { member ->
+        // Ignore deprecated fields, we don't need builders for those
+        member.annotations.none { it is Deprecated }
+      }
+      .map { it.name }
+
+  /**
+   * Validates that each field in [LeakCanary.Config] has a matching builder function
+   * in [LeakCanary.Config.Builder]
+   */
+  @Test fun `LeakCanary Config Builder matches LeakCanary Config`() {
+    val propertiesInConfig = configProperties()
+    val functionsInConfigBuilder = configBuilderFunctions()
+
+    val missingInConfigBuilder = propertiesInConfig.subtract(functionsInConfigBuilder)
+    assertThat(missingInConfigBuilder)
+        .withFailMessage("Fields in LeakCanary.Config don't have matching " +
+            "builder functions in LeakCanary.Config.Builder: $missingInConfigBuilder")
```

Comment:
instead of subtract, how about asserting the two lists are equal? That way the framework tells us what's different. If there's a reason not to use equals, then the error message should include both lists